osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs

```diff
@@ -421,13 +422,42 @@ private static void applyAnchor(Drawable drawable, Anchor anchor)
             drawable.Position -= drawable.AnchorPosition - previousAnchor;
         }
 
-        private static void applyOrigin(Drawable drawable, Anchor origin)
+        private static void applyOrigin(Drawable drawable, Anchor screenSpaceOrigin)
         {
-            if (origin == drawable.Origin) return;
+            var boundingBox = drawable.ScreenSpaceDrawQuad.AABBFloat;
 
-            var previousOrigin = drawable.OriginPosition;
-            drawable.Origin = origin;
-            drawable.Position += drawable.OriginPosition - previousOrigin;
+            var targetScreenSpacePosition = screenSpaceOrigin.PositionOnQuad(boundingBox);
+
+            Anchor localOrigin = Anchor.TopLeft;
+            float smallestDistanceFromTargetPosition = float.PositiveInfinity;
+
+            void checkOrigin(Anchor originToTest)
+            {
+                Vector2 positionToTest = drawable.ToScreenSpace(originToTest.PositionOnQuad(drawable.DrawRectangle));
+                float testedDistance = Vector2.Distance(targetScreenSpacePosition, positionToTest);
+
+                if (testedDistance < smallestDistanceFromTargetPosition)
+                {
+                    localOrigin = originToTest;
+                    smallestDistanceFromTargetPosition = testedDistance;
+                }
+            }
+
+            checkOrigin(Anchor.TopLeft);
+            checkOrigin(Anchor.TopCentre);
+            checkOrigin(Anchor.TopRight);
+
+            checkOrigin(Anchor.CentreLeft);
+            checkOrigin(Anchor.Centre);
+            checkOrigin(Anchor.CentreRight);
+
+            checkOrigin(Anchor.BottomLeft);
+            checkOrigin(Anchor.BottomCentre);
+            checkOrigin(Anchor.BottomRight);
+
+            var previousOrigin = drawable.ToParentSpace(drawable.OriginPosition);
+            drawable.Origin = localOrigin;
+            drawable.Position += drawable.ToParentSpace(drawable.OriginPosition) - previousOrigin;
```

Comment:
Taking one of my ideas from the previous comments above, here's an alternative to this that appears to fix the one frame jumps?

```diff
diff --git a/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs b/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
index 21909cdc10..8fd9c1b559 100644
--- a/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
+++ b/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
@@ -455,9 +455,10 @@ void checkOrigin(Anchor originToTest)
             checkOrigin(Anchor.BottomCentre);
             checkOrigin(Anchor.BottomRight);

-            var previousOrigin = drawable.ToParentSpace(drawable.OriginPosition);
+            Vector2 offset = drawable.ToParentSpace(localOrigin.PositionOnQuad(drawable.DrawRectangle)) - drawable.ToParentSpace(drawable.Origin.PositionOnQuad(drawable.DrawRectangle));
+
             drawable.Origin = localOrigin;
-            drawable.Position += drawable.ToParentSpace(drawable.OriginPosition) - previousOrigin;
+            drawable.Position += offset;
         }

         private static void adjustScaleFromAnchor(ref Vector2 scale, Anchor reference)
```

i.e:
- Get current the position of our origin in the parent.
- Get the new origin position in the parent.
- Perform the subtraction on this. It's important that this is done in the parent space so that it considers rotation.

Reply:
Is this still valid for consideration when reviewing this change? Not sure if this was posted before or after the approval.

Reply:
The above was after the approval. In principle I don't see any reason to not apply the change above but I haven't tested yet.